src/recipe/recipe.service.ts

```diff
@@ -77,10 +77,10 @@ export class RecipeService {
 
   async fetchAllRecipes(userId: number): Promise<Recipe[]> {
     const user = await this.userRepository.getUserById(userId);
-    if (user.role === Role.ADMIN) {
+    if (user?.role === Role.ADMIN) {
```

Comment:
why user might not have `role` defined?

Reply:
Thats interesting because I accidentally made a request with a token from a user who has been deleted (I created user > sign in > `prisma migrate reset` > fetch with the old token). This caused 500 error which was caused by the fact, that `undefined` doesn't have `role` field.